src/Telegram/Types/ChatMember.php

```diff
@@ -149,4 +52,30 @@ public function mapSubObjects(string $key, array $data): TelegramTypes
 
         return parent::mapSubObjects($key, $data);
     }
+
+    /**
+     * Create specific ChatMember class based on 'status'. If there is no match,
+     */
+    public static function create(array $data, LoggerInterface $logger): ChatMember
+    {
+        $status = $data['status'] ?? null;
+
+        foreach (self::CHILD_CLASS_REFERENCES as $childClassReference) {
+            if ($status === $childClassReference::STATUS) {
+                return new $childClassReference($data, $logger);
+            }
+        }
+        $logger->error(sprintf(
+            'Unable to detect correct "%s" class based on status = "%s"! Maybe a recent Telegram Bot ' .
+            'API update? In any way, please submit an issue (bug report) at %s with this complete log line',
+            self::class,
+            $status ?? 'null',
+            'https://github.com/unreal4u/telegram-api/issues'
+        ), [
+            'object' => self::class,
+            'data' => $data,
+        ]);
+
+        return new ChatMember($data, $logger);
```

Comment:
It's ok to throw an exception here. I would not support creating an object that is invalid to begin with.